smart-pix/smart-pixels-ml
# Add Documentation, Smoke Tests, and requirements.txt

1. **Documentation** 
 - Added documentation files in the `docs/` directory, written in Markdown and fully compatible with MkDocs.
 - Added a `mkdocs.yml` file to configure the MkDocs site.
 - Generated the static `site/` directory using:
    ```bash
    mkdocs build
    ```
    The `site/` directory contains the static HTML files for the documentation site.

2. **Smoke Tests**:
 - Implemented smoke tests to validate the basic setup, data generation, and model training functionalities as in `test_run.py`.
 - The smoke test script can be run using:
    ```bash
    python test_run.py
    ```
    - The smoke tests can be moved to the `tests/` directory once more tests are added.

3. Requirements file: `requirements.txt` has been added with the necessary dependencies.
Can be installed using:
    ```bash
    pip install -r requirements.txt
    ```

### How to Test
0. Install the dependencies: `pip install -r requirements.txt`.
1. Run the smoke test script: `python test_run.py`.
2. View the documentation locally: `mkdocs serve`.
3. Check the site build: `mkdocs build`.
